tfhe/src/shortint/oprf.rs

```diff
@@ -0,0 +1,191 @@
+use super::Ciphertext;
+use crate::core_crypto::commons::math::random::RandomGenerator;
+use crate::core_crypto::prelude::{
+    lwe_ciphertext_plaintext_add_assign, ActivatedRandomGenerator, Plaintext,
+};
+use crate::shortint::ciphertext::Degree;
+use crate::shortint::server_key::LookupTableOwned;
+use crate::shortint::ServerKey;
+use concrete_csprng::seeders::Seed;
+
+impl ServerKey {
+    fn create_random_from_seed(&self, seed: Seed) -> Ciphertext {
+        let mut ct = self.create_trivial(0);
+
+        let mut generator = RandomGenerator::<ActivatedRandomGenerator>::new(seed);
+
+        for mask_e in ct.ct.get_mut_mask().as_mut() {
+            *mask_e = generator.random_uniform::<u64>();
+        }
+
+        ct
+    }
+
+    /// Uniformly generates a random value in `[0, 2^random_bits_count[`
+    /// `2^random_bits_count` must be smaller than the message modulus
+    pub fn generate_oblivious_pseudo_random(
+        &self,
+        seed: Seed,
+        random_bits_count: u64,
+    ) -> Ciphertext {
+        assert!(1 << random_bits_count < self.message_modulus.0);
+
+        self.generate_oblivious_pseudo_random_message_and_carry(seed, random_bits_count)
+    }
+
+    /// Uniformly generates a random value in `[0, 2^random_bits_count[`
+    pub fn generate_oblivious_pseudo_random_message_and_carry(
```

Comment:
I think you said it was maybe not useful after all ?

Reply:
Yes, it does not seem very useful, but it makes generate_oblivious_pseudo_random a one-liner so it does not cost much to have

Reply:
ok, maybe pub(crate) unsure

Reply:
Me neither

Reply:
Done